torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,428 @@
+    for more information about the dataset.
+
+    Dataset features:
+
+    * infrared channel images from the Himawari weather satellite (512x512 px)
+      at 5km spatial resolution
+    * auxiliary features such as wind speed, pressure, and more that can be used
+      for regression or classification tasks
+    * 1,099 typhoons and 189,364 image
+
+    Dataset format:
+
+    * hdf5 files containing the infrared channel images
+    * .csv files containing the metadata for each image
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.20783/DIAS.664
+
+    .. versionadded:: 0.6
+    """
+
+    valid_tasks = ["classification", "regression"]
+    aux_file_name = "aux_data.csv"
+
+    valid_features = [
+        "year",
+        "month",
+        "day",
+        "hour",
+        "grade",
+        "lat",
+        "lng",
+        "pressure",
+        "wind",
+        "dir50",
+        "long50",
+        "short50",
+        "dir30",
+        "long30",
+        "short30",
+        "landfall",
+        "intp",
+    ]
+
+    url = "https://huggingface.co/datasets/torchgeo/digital_typhoon/resolve/main/WP.tar.gz{0}"  # noqa: E501
+
+    md5sums = {
+        "aa": "3af98052aed17e0ddb1e94caca2582e2",
+        "ab": "2c5d25455ac8aef1de33fe6456ab2c8d",
+    }
+
+    data_root = "WP"
+
+    def __init__(
+        self,
+        root: str = "data",
+        task: str = "regression",
+        features: Sequence[str] = ["wind"],
+        target: list[str] = ["wind"],
+        sequence_length: int = 3,
+        min_feature_value: Optional[dict[str, float]] = None,
+        max_feature_value: Optional[dict[str, float]] = None,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Digital Typhoon Analysis dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            task: whether to load "regression" or "classification" labels
+            features: which auxiliary features to return
+            target: which auxiliary features to use as targets
+            sequence_length: length of the sequence to return
+            min_feature_value: minimum value for each feature
+            max_feature_value: maximum value for each feature
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``task`` argument is invalid
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        try:
+            import h5py  # noqa: F401
```

Comment:
@adamjstewart this is still being done in several datasets (chabud, skippd, so2sat, zuericrop) but `requirements/datasets.txt`lists h5py as a dependency. However, when I don't do the import in the init but at the top of the file, the docs fail with `no module h5py`

Reply:
besides this and the pesky mypy errors, this should be ready for review

Reply:
`requirements/required.txt` is required deps that must always be installed, `requirements/datasets.txt` is optional deps that may or may not be installed.

Reply:
ah duuh, thanks!